examples/piecestore/rpc/client/main.go

```diff
@@ -0,0 +1,183 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/urfave/cli"
+	"github.com/zeebo/errs"
+
+	"golang.org/x/net/context"
+
+	"google.golang.org/grpc"
+
+	"storj.io/storj/examples/piecestore/rpc/client/api"
+	"storj.io/storj/examples/piecestore/rpc/client/utils"
+	pb "storj.io/storj/protos/piecestore"
+)
+
+var argError = errs.Class("argError")
+
+func main() {
+
+	app := cli.NewApp()
+
+	// Set up connection with rpc server
+	var conn *grpc.ClientConn
+	conn, err := grpc.Dial(":7777", grpc.WithInsecure())
+	if err != nil {
+		log.Fatalf("did not connect: %s", err)
+	}
+	defer conn.Close()
+	client := pb.NewPieceStoreRoutesClient(conn)
+	ctx := context.Background()
+
+	app.Commands = []cli.Command{
+		{
+			Name:    "upload",
+			Aliases: []string{"u"},
+			Usage:   "upload data",
+			Action: func(c *cli.Context) error {
+				if c.Args().Get(0) == "" {
+					return argError.New("No input file specified")
+				}
+
+				file, err := os.Open(c.Args().Get(0))
+				if err != nil {
+					return err
+				}
+				// Close the file when we are done
+				defer file.Close()
+
+				fileInfo, err := os.Stat(c.Args().Get(0))
```

Comment:
missing err check. i swear i've left this comment before lol

Reply:
also another comment i made before is that you can just do file.Stat() which might be faster since some of the data should already be cached in the kernel's file descriptor

Reply:
👍